action.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+set -euo pipefail
+
+# Repeat the character 80 times
+long_line=$(printf '―%.0s' {1..80})
+bold=$(echo -e "\033[1m")
+reset=$(echo -e "\033[0m")
+
+echo "$long_line"
+echo "${bold}→ Preparing${reset}"
+echo
+echo "Checking out commit"
+echo "https://github.com/$GITHUB_REPOSITORY/commit/$GITHUB_SHA"
+echo
+
+if [[ -z "$GITHUB_TOKEN" ]]; then
+  auth_header=''
+else
+  auth_header="authorization: Bearer $GITHUB_TOKEN"
+fi
+curl \
+  --silent \
+  --header "$auth_header" \
+  -L https://api.github.com/repos/$GITHUB_REPOSITORY/tarball/$GITHUB_SHA \
+  | \
+  tar --strip-components=1 -xzf -
```

Comment:
I'm confused - I thought the checkout step was supposed to get rid of this bit?

Reply:
Ah, yes. I initially thought that the checkout step was the one thing standing in the way of us rewriting the action in bash. So my initial proposal was to pull that out into a separate step run before this action. That would be a breaking change involving a switch to `v2` and, eventually, updating every project's workflow. I think this would have been worth it in any case, but it turned out to be pretty trivial to do the checkout in bash so this allowed us to avoid a breaking change.

Reply:
I've just realised one possible source of confusion. The [.github/workflow/tests.yml](https://github.com/opensafely-core/research-action/blob/b94d50287367e9e566a32135636f7167193ee350/.github/workflows/tests.yml) file defines how this action tests itself, it's not what defines the behaviour of this action which is done by the [action.yml](https://github.com/opensafely-core/research-action/blob/b94d50287367e9e566a32135636f7167193ee350/action.yml) file.